napalm_ios/ios.py

```diff
@@ -134,14 +135,18 @@ def _send_command(self, command):
 
         If command is a list will iterate through commands until valid command.
         """
-        if isinstance(command, list):
-            for cmd in command:
-                output = self.device.send_command(cmd)
-                if "% Invalid" not in output:
-                    break
-        else:
-            output = self.device.send_command(command)
-        return self._send_command_postprocess(output)
+        try:
+            if isinstance(command, list):
+                for cmd in command:
+                    output = self.device.send_command(cmd)
+                    if "% Invalid" not in output:
+                        break
+            else:
+                output = self.device.send_command(command)
+            return self._send_command_postprocess(output)
+        except (socket.error, EOFError):
+            self.open()
+            return self._send_command(command)
```

Comment:
The only thing that worries me here is a very corner case when for some reason you cannot access the device (sudden DNS issue, ISP problem, or whatever reason the channel cannot be established), and we could go into an infinite loop.
But I understand that `open` would raise an error when unable to access the device anymore and it would stop there? (I am speaking without extensive experience with IOS, but I would like to hear the explanation on this and make sure we don't introduce another bug).